Prep for Styling Update

JohannesSeikowsky opened the pull request:
This is just some prep for adding the Custom CSS that my friend Marcello wrote for this page. Mostly setting relevant IDs and some other details.
I'll add the actual CSS in a separate PR in the gutenbergsite repo (where the CSS files live).

JohannesSeikowsky commented:
also deleting the QR code (agreed with Greg)

eshellman commented on templates/bibrec.html:
```diff
@@ -185,7 +180,7 @@ <h2>Read now or download (free!)</h2>
                   class="unpadded icon_save"
                   ><a href="/${file_.filename}" type="${file_.mediatypes[-1]}" class="link"
-                      title="${'Read online' if file_.filetype and file_.filetype.startswith('html') else 'Download'}">${file_.hr_filetype}</a></td>
+                      title="${'Read online' if file_.filetype and file_.filetype.startswith('html') else 'Download'}" id="${'read_online_link' if file_.filetype and file_.filetype.startswith('html') else ''}">${'Read online now' if 'Read online' in file_.hr_filetype else file_.hr_filetype}</a></td>
```
can you describe what `${'Read online now' if 'Read online' in file_.hr_filetype else file_.hr_filetype}` does?
and why?

JohannesSeikowsky replied:
That is to change the text of link from "Read online (web)" to "Read online now". Maybe you can think of a better way of doing it?

eshellman replied:
I think that comes from the database

eshellman reviewed (commented):
are we turning off the QRCode page as well?

eshellman commented:
The easiest way to test css is to put it inline in a test branch in this repo

JohannesSeikowsky commented:
I've added the CSS inline for testing as you suggested (I added it in the place where in would be loaded).
I'm not sure about turning off the QR code page - I'll drop Greg a quick message about that ...

JohannesSeikowsky commented:
I suppose the next step now would be for you to test it in your environment (in mine it seems ok )? or test in dev?

eshellman commented:
I need to sort the css update first

JohannesSeikowsky commented:
I've just heared back from Greg about the QR codes. He writes:
"I’d be happy to turn off the QR code."

eshellman commented:
Comments also on https://github.com/gutenbergtools/gutenbergsite/pull/47

- too much space between header and breadcrumbs
- to much space between breadcrumbs and title
- when wrapped, the line spacing in breadcrumbs could be tighter and following 2nd line indented
- when wrapped, the line spacing in title is too tighter
- I never liked the light brown title, black is easer to read
- How about making presenting the author string distinguishable from the title?
- How about putting a soft break between author string and title: `<wbr/>`
- I think the "Read online now" button is too hard to see. How about making the text look more like the other links? 
- Also how about just "Read now" ? or "Read now!" ?
- On the QR Code. Good riddance on the small screen, but it's useful on the desktop because it makes it easier to transfer to your phone when you want to take to bed to read. Also, the empty space is noticable on desktop. People will complain if they've been using it and it goes away.
- nice touch making the table header darker.
- on iphone view, "Choose how to read this book" takes a lot of space. How about "How to read"?

eshellman commented on templates/bibrec.html:
```diff
@@ -184,8 +274,10 @@ <h2>Read now or download (free!)</h2>
+                  ><a href="/${file_.filename}" type="${file_.mediatypes[-1]}"
+                      class="link ${'read_html' if file_.filetype and file_.filetype.startswith('html') else ''}"
+                      title="${'Read online' if file_.filetype and file_.filetype.startswith('html') else 'Download'}"
+                      id="">${file_.hr_filetype}</a></td>
```
```suggestion
                      >${file_.hr_filetype}</a></td>
```
ids can't be empty

eshellman commented on templates/bibrec.html:
```diff
@@ -222,16 +314,18 @@ <h2>Read now or download (free!)</h2>
                  charset="${e.charset}"
-                 accesskey="${str (n % 10)}">
+                 accesskey="${str (n % 10)}"
+                 id="similar-books-link">
```
```suggestion
                 class="similar-books-link">
```
can't use ids because this is a repeated element

eshellman commented on templates/bibrec.html:
```diff
@@ -222,16 +314,18 @@ <h2>Read now or download (free!)</h2>
           <py:if test="isinstance (e, bs.Cat) and e.rel not in ('start', )">
             <py:if test="e.header and old_header != e.header">
-              <h2 class="header">${e.header}</h2>
+              <h2 class="header" id="middle_section_header">${e.header}</h2>
```
check that this only occurs once, otherwise use a class selector

JohannesSeikowsky replied:
I'll make it a class. might as well

eshellman reviewed (commented):
found id related errors in nu validator

eshellman commented on templates/bibrec.html:
```diff
@@ -52,15 +52,107 @@
+	#middle_section_header,
+	#about_book_header,
+	#download_options_header {
+	    color: black;
```
here, you're over-riding a style selected by `.page_content h2`
should we just change that style for the whole site for consistency?

JohannesSeikowsky replied:
obviously if you have the time/inclination you can verify that this wouldn't create unintended consequences and commit the change. As said I'm generally not again it at all ...

eshellman commented on templates/bibrec.html:
```diff
@@ -52,15 +52,107 @@
+	    color: black;
+	    margin-top: 52px;
+	    font-size: 24px;
+	}
```
here you're using absolute sizes. for accessibility, it's better to use relative sizes, especially for font sizes, to let users with poor vision change the base font size.

eshellman commented on templates/bibrec.html:
```diff
@@ -52,15 +52,107 @@
+	@media screen and (max-width: 600px) {
+	    #book_title {
+		font-size: 24px;
+	    }
```
here you're using absolute sizes. for accessibility, it's better to use relative sizes, especially for font sizes, to let users with poor vision change the base font size.

eshellman commented on templates/bibrec.html:
```diff
@@ -52,15 +52,107 @@
+	    padding-bottom: 12px;
+	    margin: 48px 0 36px;
+	    font-size: 32px;
+	}
```
here you're using absolute sizes. for accessibility, it's better to use relative sizes, especially for font sizes, to let users with poor vision change the base font size.
Should we just modify the h1 style for the whole site (maybe except for the border_bottom?)

eshellman commented on templates/bibrec.html:
```diff
@@ -52,15 +52,107 @@
+	    transition: all 0.3s ease;
+	    font-size: 16px;
+	    font-weight: bold;
+	}
```
consider using relative sizes

eshellman reviewed (commented):
some of the changes here should be made on the whole site or not at all.
avoid absolute sizes for accessibility.

eshellman commented:
Overall impression: really great work; but it's important to work towards simplification of the sitewide CSS.

JohannesSeikowsky commented:
I've committed the specific code changes you suggested. Now going through the general suggestion you made to see which ones can be implement right away (I pretty much agree with all of them)

JohannesSeikowsky commented:
I've gone through your suggestions and committed a few changes.
Here my comments on the various points:


too much space between header and breadcrumbs
- I've tried it. reducing that space makes the page look cramped.

to much space between breadcrumbs and title
- I've tried it. good point. I've reducing it a bit and it looks more "compact" now. reducing more makes it look crowded.

when wrapped, the line spacing in breadcrumbs could be tighter and following 2nd line indented
- agreed. I've made the spacing tighter. feel free to take care of the indentation if you think it very important.

when wrapped, the line spacing in title is too tighter
- agreed. I've increased the spacing.

How about making presenting the author string distinguishable from the title?
- I generally agree. but maybe better in a separate PR/deployment. There're already a good amount of changes in here.
Will also require some additional visual design work to make it look good. will need to consult w Marcello.

How about putting a soft break between author string and title: <wbr/>
- see above

I think the "Read online now" button is too hard to see. How about making the text look more like the other links?
- agreed. I've made an attempt. May not be perfect, but best I could figure out for now. 

Also how about just "Read now" ? or "Read now!" ?
- agreed and done

On the QR Code. 
- agreed. back in but only visible on larger screens.

on iphone view, "Choose how to read this book" takes a lot of space. How about "How to read"?
- agreed and done

JohannesSeikowsky commented:
Unless there are any active bugs I think we should aim to deploy fairly soon ...

JohannesSeikowsky commented:
This is how it looks for me now locally. I'm not loading the icon images and qr code, but they would obviously be there in dev and production.


![Screenshot_2025-04-13_15-41-34](https://github.com/user-attachments/assets/63ca8c95-a426-432e-a69c-729f5f30725d)
![Screenshot_2025-04-13_15-41-43](https://github.com/user-attachments/assets/91d101fd-e6bf-4fb1-9a5a-ab838a168f74)

eshellman commented:
If you put the css in gutenberg site, I'll can push the relevant branch to dev.gutenberg.org

JohannesSeikowsky commented:
sounds good. I've two more lessons to teach now, will do right after

eshellman commented:
I've now set up machinery so that the dev branch of the gutenbergsite repo can get tracked by dev.gutenberg.org, making it much easier to test the new css! So please target the dev branch in that repo in your PR for the static pages. When you've removed the css changes from this PR, I will merge it to master and run on dev. Thanks for giving us impetus to make the dev process changes!

JohannesSeikowsky commented:
That's great!! I've copy-pasted the CSS from here to the bottom of style2.css in the dev branch of the gutenbergsite repo and committed it. took a little longer than I planned - the wrapping and size of the "read now!" button on mobile was not good and I spent some time trying to get it right.

JohannesSeikowsky commented:
does dev.gutenberg.org continually sync with the dev branch of the gutenbergsite repo? or is there a time lag?

eshellman commented:
I'm working on it

eshellman commented:
It's live https://dev.gutenberg.org/ebooks/75387

JohannesSeikowsky commented:
Looks good at first glance, but the "read now" button still says "read online now" rather than "Read now!". Any idea why that is?

JohannesSeikowsky commented:
that longer text can actually cause some trouble on small screen sizes. Besides I think just "Read now!" is just better wording.

JohannesSeikowsky commented:
I also just noticed that the last row of the Download options table that says "There may be more files related to this item" is a bit wider than the other rows for some reason.
Should we maybe just take that row out? I'm not sure there's so much value in having it ...

srjfoo commented:
> I also just noticed that the last row of the Download options table that says "There may be more files related to this item" is a bit wider than the other rows for some reason. Should we maybe just take that row out? I'm not sure there's so much value in having it ...

Please don't take it out unless you can restrict the removal to small devices like phones.

JohannesSeikowsky commented:
ok, but why?

srjfoo commented:
Admittedly, I'm not the typical user -- I'm a siteadmin with Distributed Proofreaders, and there are times that we do need to look back at older files.

srjfoo commented:
> makes sense. that row is literally just hardcoded text though - not a link or anything like that. And it's on every page. So I'm not sure how that's helpful for the typical user or for yourself ...

If you look very closely, "more files" has a link associated with it.

eshellman commented:
> Looks good at first glance, but the "read now" button still says "read online now" rather than "Read now!". Any idea why that is?

that would be in the autocat3 template

eshellman commented:
> > makes sense. that row is literally just hardcoded text though - not a link or anything like that. And it's on every page. So I'm not sure how that's helpful for the typical user or for yourself ...
> 
> If you look very closely, "more files" has a link associated with it.

it's intentionally hidden. I did that! We switched to using generated html5 files 2 years ago, and that link makes it easier to look at the submitted file when something goes wrong. Whe you converting 75,000 files, there are inevitably things that go wrong.

JohannesSeikowsky commented:
ok, makes sense. Didn't know ...

JohannesSeikowsky commented:
> > Looks good at first glance, but the "read now" button still says "read online now" rather than "Read now!". Any idea why that is?
> 
> that would be in the autocat3 template

yep. I thought that was already in there. Looking into it now ...

JohannesSeikowsky commented:
@eshellman changed it and made a new pull request as this one is already merged.
